tests/php/ORM/DataObjectTest.php

```diff
@@ -2212,7 +2215,10 @@ public function testBelongsToPolymorphic()
 
         // Write object with components
         $ceo->write(false, false, false, true);
-        $this->assertTrue($ceo->CompanyOwned()->isInDB(), 'write() writes belongs_to components to the database.');
+        $this->assertFalse(
+            $ceo->CompanyOwned()->isInDB(),
+            'write() does not write belongs_to components to the database that do not already exist.'
+        );
 
         $newCEO = DataObject::get_by_id(DataObjectTest\CEO::class, $ceo->ID);
```

Comment:
Some of these other tests assumes that the CompanyOwned will have been written. I guess those tests are still passing because they're implicitly creating a new Company instance without an ID. However, this is not intuitive and the test should be updated to reflect that new nuance.

This apply to both testBelongsToPolymorphic and testBelongsTo.

If the intention is that people will manually call write on Company Owned, than we should have a test to cover this use case as well. Maybe some of the existing assertion should be moved to this scenario.

Reply:
> I did the original PR against the 4.3 branch. My guess is that this will break in 4.3 as well. This PR should target 4.3 as well.

I'll change it to target 4.3

> I think we should add some test cases to cover the scenario where the developer manually triggers the write on the component. It's arguably a bit picky, but this is going out on patch releases for 4.3 or, 4.4 as well. I think we need to be extra careful.

I added a simple test to make sure this works.

> Some of these other tests assumes that the CompanyOwned will have been written. I guess those tests are still passing because they're implicitly creating a new Company instance without an ID. However, this is not intuitive and the test should be updated to reflect that new nuance.
> 
> This apply to both testBelongsToPolymorphic and testBelongsTo.
> 
> If the intention is that people will manually call write on Company Owned, than we should have a test to cover this use case as well. Maybe some of the existing assertion should be moved to this scenario.

If you're talking about accessing `CompanyOwned`, that will always return an object.